app/src/context/projects-context.tsx

```diff
@@ -28,34 +32,67 @@ export function ProjectsProvider({children}: {children: ReactNode}) {
   const [projects, setProjects] = useState<ProjectExtended[]>([]);
 
   useEffect(() => {
-    const init = async () => {
-      const localProjects = await getProjectsDB();
+    initProjects();
+  }, []);
+
+  /**
+   * Initializes the application by synchronizing local and remote projects.
+   * First, it fetches projects from the local database and sets them in the project state,
+   * then it synchronizes with remote projects to update the local state and database as needed.
+   *
+   * @async
+   * @function init
+   * @returns {Promise<void>} Resolves when initialization is complete.
+   */
+  const initProjects = async () => {
+    await syncLocalProjects();
+    await syncRemoteProjects();
+  };
 
-      setProjects(localProjects);
+  /**
+   * Synchronizes remote projects with the local state and database.
+   * Fetches projects from a remote source, merges with the local project data
+   * to retain activation and sync status, then updates the project state and database.
+   *
+   * @async
+   * @function syncRemoteProjects
+   * @returns {Promise<void>} Resolves when synchronization is complete.
+   */
+  const syncRemoteProjects = async () => {
+    const remoteProjects = await getRemoteProjects();
 
-      const remoteProjects = await getRemoteProjects();
+    if (!remoteProjects.length) return;
 
-      if (!remoteProjects.length) return;
+    const currentProjectsMap = new Map(
+      projects.map(project => [project._id, project])
+    );
 
-      const localProjectsMap = new Map(
-        localProjects.map(project => [project._id, project])
-      );
+    const newProjects = remoteProjects.map(project => {
```

Comment:
so you are mapping only across the remote projects, this means that any projects which were removed online will be no longer available on the device? Is this desired behaviour. What if unsynced changes?